### PR TITLE
[CELEBORN-305] Change the parameter passed in the registerShuffle method to numPartitions instead of numMappers

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### What changes were proposed in this pull request?

Change the parameter passed in the registerShuffle method to numPartitions instead of numMappers

### Why are the changes needed?

ShuffleClientImpl's registerShuffle method should pass numPartitions instead of numMappers

### Does this PR introduce _any_ user-facing change?

NO

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 1]
```diff
@@ -313,7 +313,7 @@ private ConcurrentHashMap<Integer, PartitionLocation> registerShuffle(
     return registerShuffleInternal(
         shuffleId,
         numMappers,
-        numMappers,
```
Comment (RexXiong) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:316]: Good catch!  Although num partitions is actually correctly used when register shuffle

Comment (jiaoqingbo) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:316]: Yes, currently this parameter is only used in printing logs

**Comment by FMX** [comment]
That UT is not stable yet.

**Review by FMX — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**